doc/source/object-store.rst

```diff
@@ -0,0 +1,169 @@
+Object Store (Plasma)
+=====================
+
+Plasma is an in-memory object store that is being developed as part of `Apache Arrow`_. Ray uses Plasma to efficiently transfer objects across different processes and different nodes.
+
+All objects in Plasma object store are **immutable** and held in shared memory. This is so that they can be accessed efficiently by many workers on the same node.
+
+Note that Plasma does not support all data-types, and in certain cases it may be necessary to either write your own serialization protocol or use Actors to hold objects and transfer object state (i.e., weight matrices) among Ray workers.
```

Comment:
```suggestion
Note that Plasma does not support all data-types, and in certain cases it may be necessary to either write your own serialization protocol using.... TODO LINK TO DOCUMENTATION FOR ray.register_custom_serializer
```